akka-http-core/src/main/scala/akka/http/impl/engine/client/pool/NewHostConnectionPool.scala

```diff
@@ -325,10 +327,18 @@ private[client] object NewHostConnectionPool {
 
           def settings: ConnectionPoolSettings = _settings
 
+          private lazy val keepAliveDurationFuzziness: () ⇒ Long = {
+            val random = new Random()
```

Comment:
This constructs a new `Random` instance for each connection - could we do that less often?

Reply:
this random is created once per slot.
And slots are created for the pool here _once_ 
https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/impl/engine/client/pool/NewHostConnectionPool.scala#L70

Reply:
ah, you're right, the `() ⇒ random.nextLong() % max` is created below, I misread :+1: